pontoon/administration/views.py

```diff
@@ -164,16 +168,29 @@ def manage_project(request, slug=None, template="admin_project.html"):
                 for locale in locales:
                     ProjectLocale.objects.get_or_create(project=project, locale=locale)
 
+                project_locales = ProjectLocale.objects.filter(project=project)
+
                 # Update readonly flags
                 locales_readonly_pks = [l.pk for l in locales_readonly_form]
-                project_locales = ProjectLocale.objects.filter(project=project)
                 project_locales.exclude(locale__pk__in=locales_readonly_pks).update(
                     readonly=False
                 )
                 project_locales.filter(
                     locale__pk__in=locales_readonly_pks,
                 ).update(readonly=True)
 
+                # Update pretranslate flags
+                locales_pretranslate_form = form.cleaned_data.get(
+                    "locales_pretranslate", []
+                )
+                locales_pretranslate_pks = [l.pk for l in locales_pretranslate_form]
+                project_locales.exclude(locale__pk__in=locales_pretranslate_pks).update(
+                    pretranslation_enabled=False
+                )
+                project_locales.filter(
+                    locale__pk__in=locales_pretranslate_pks,
+                ).update(pretranslation_enabled=True)
```

Comment:
non-blocking, nitpick: This _feels_ like it's doing two loops where one might be enough. But optimising this almost certainly isn't worthwhile.

Also, is the code style formatter doing something weird here as these two calls place newlines in different places?

Reply:
Ah, there's an extra comma.

Reply:
Re the optimization - how can I get to the same result with just one SQL UPDATE?

Reply:
I could figure out the SQL for that (the assigned value can be an expression, not just a constant), but I'm not sure how to get there through Django's ORM.

Reply:
Makes sense. I'd opt for sticking with Django ORM unless absolutely necessary.

Reply:
Agreed.